src/serverless-openapi-typescript.ts

```diff
@@ -183,8 +184,23 @@ export default class ServerlessOpenapiTypeScript {
         const outputFile = this.serverless.processedInput.options.output;
         const openApi = yaml.load(fs.readFileSync(outputFile));
         this.patchOpenApiVersion(openApi);
-        this.tagMethods(openApi);
-        fs.writeFileSync(outputFile, outputFile.endsWith('json') ? JSON.stringify(openApi, null, 2) : yaml.dump(openApi));
+        this.enrichMethodsInfo(openApi);
+        const encodedOpenAPI = this.encodeOpenApiToStandard(openApi);
+        fs.writeFileSync(outputFile, outputFile.endsWith('json') ? JSON.stringify(encodedOpenAPI, null, 2) : yaml.dump(encodedOpenAPI));
+    }
+
+    encodeOpenApiToStandard(openApi) {
+        const INVALID_CHARACTERS_KEY = /<|>/g;
+        const INVALID_CHARACTERS_REF = /%3C|%3E/g;
+
+        const mapObject = mapKeysDeep(openApi, (value, key) =>
+            INVALID_CHARACTERS_KEY.test(key) ? key.replace(INVALID_CHARACTERS_KEY, '_') : key
+        );
+
+        return mapValuesDeep(mapObject, (value, key) =>
+            isString(value) && key === '$ref' && INVALID_CHARACTERS_REF.test(value) ?
```

Comment:
Matter of personal of style so your call - I think the interim value of `mapObject` does nothing but make this method a little more verbose than it needs to be.  
Of course it makes sense to avoid nested functions like `f(g(x))`.  

Deepdash, like lodash does offer a way to chain functions which make it easier to follow the functional path:  

```typescript
_(openApi)
  .mapKeysDeep(stubGenerics)
  .mapValuesDeep(stubBase64EncodedGenerics)
  .value();
```

Reply:
I didn't see they offer this chain - they offer to put the functions on the lodash object `_` but I think not in the chain (more complex to do it) But I will try

Reply:
I don't like how they mix it with lodash and it will not work with the chain also we have typescript problem. 

```
// or cherry-pick method you only need and mix it into lodash
require('deepdash/addFilterDeep')(_);
// or cherry-pick method separately if you don't want to mutate Lodash instance
const filterDeep = require('deepdash/getFilterDeep')(_);
```

I tried to do something with flow - but it just complex it and increase the code